src/components/Wizard/BlueprintWizard.js

```diff
@@ -87,6 +87,20 @@ const BlueprintWizard = (props) => {
         ? formValues
         : formApi.getState().values;
     const blueprintData = formStateToBlueprint(formState);
+
+    // when customizations are provided, the users and filesystem customizations
+    // are initialized to an empty array. This creates issues for image types that
+    // aren't allowed these customizations
+    if (blueprintData.customizations) {
+      if (blueprintData.customizations.filesystem?.length === 0) {
+        delete blueprintData.customizations.filesystem;
+      }
+
+      if (blueprintData.customizations.user?.length === 0) {
```

Comment:
We may actually need to check some other fields such as services, locale etc if the user has stepped through those form steps, some of those values might be empty too.

Reply:
Hmm I've checked the other steps and we get this when I click through each step but don't edit anything:

```
[customizations]
[customizations.kernel]
append = ""
[customizations.timezone]
[customizations.locale]
[customizations.firewall]
[customizations.firewall.services]
[customizations.services]
[customizations.fdo]
[customizations.ignition]
[customizations.ignition.firstboot]
```

However, it seems like the only one that was causing issues was the filesystem customizations.